src/bitset.hpp

```diff
@@ -169,10 +188,33 @@ class Bitset {
   // (e.g. '11000011' would return '000011110011').
   static Bitset PCSPOfRootsplit(const Bitset &rootsplit);
 
+  // NNI: Nearest Neighbor Interchange
+  // Given parent/child pair of subsplits embedded within a subsplit DAG, produces the
+  // resulting parent/child subsplit pair that result from an NNI operation.
+  // Get the data.
+  static BitsetPair SubsplitNNI(const Bitset &parent_in, const Bitset &child_in,
+                                const size_t swap_which_child_with_sister,
+                                const size_t which_child_of_parent);
+  static BitsetPair SubsplitNNI(const Bitset &parent_in, const Bitset &child_in,
+                                const size_t swap_which_child_with_sister);
+  // Get data value.
+  std::vector<bool> GetValue() const;
+  // Comparator operator. Returns `1 if this > that`, `-1 if that > this` and 0 if `this
+  // == that`.
+  int Compare(const Bitset &that) const;
+
  private:
   std::vector<bool> value_;
 };
 
+class TaxonSet : public Bitset {};
```

Comment:
I'm guessing these are left-overs?

Reply:
This was something I was going to propose to you.  We have all these functions that are prefixed "Subsplit" and "PCSP".  Would it be better to have Bitset be the base class for these instead?

Reply:
Ah, cool!

@ognian- had a really nice idea about treating these as "views" of an underlying bitset object. I can't remember if there was any committed code for it though.

Reply:
Removed these.  Maybe the subject of another issue if we want to do this?  
Also, not sure what a "view" is?